.github/workflows/build_docker_image.yml

```diff
@@ -1,64 +1,81 @@
-          TAGS="$TAGS,ghcr.io/${DOCKER_IMAGE}:latest"
-          TAGS="$TAGS,ghcr.io/${DOCKER_IMAGE}:git-${GITHUB_SHA::8}"
-          echo ::set-output name=tags::${TAGS}
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Login to DockerHub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          file: ./Dockerfile
-          build-args: ${{ matrix.build-args }}
-          platforms: linux/amd64,linux/386
-          push: true
-          tags: ${{ steps.tags.outputs.tags }}
-          labels: |
-            org.opencontainers.image.revision=${{ github.sha }}
-            org.opencontainers.image.created=${{ steps.timestamp.outputs.timestamp }}
+          docker build --build-arg buildparams="${{ matrix.build-args }}" -t ${{ matrix.tag }} . \
+            --label "org.opencontainers.image.revision=${{ github.sha }}" \
+            --label "org.opencontainers.image.created=$(date --rfc-3339=seconds --utc)"
+      - name: Test Docker image
+        run: |
+          cd samples
+          for dir in */
+          do
+            cd "$dir"
+            docker run --rm -v `pwd`:/usr/src/app -t ${{ matrix.tag }} make -j`nproc`
+            cd ..
+          done
+      - name: Push container if on master branch
+        if: github.ref == 'refs/heads/master'
+        run: |
+          docker login -u "${{ github.actor }}" -p "${{ secrets.GITHUB_TOKEN }}" ghcr.io
+          docker tag ${{ matrix.tag }} ghcr.io/${{ github.repository }}:${{ matrix.tag }}
+          docker push ghcr.io/${{ github.repository }}:${{ matrix.tag }}
+
+  xboxdev_containers:
+    name: Push Official Containers
+    if: github.repository_owner == 'XboxDev' && github.ref == 'refs/heads/master'
+    needs: [main]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [latest, debug, lto]
+    steps:
+      - name: Pull container
+        run: |
+          docker login -u "${{ github.actor }}" -p "${{ secrets.GITHUB_TOKEN }}" ghcr.io
+          docker pull ghcr.io/${{ github.repository }}:${{ matrix.tag }}
+      - name: Push to XboxDev group
+        run: |
+          if [ "${{ matrix.tag }}" == "latest" ]; then
+            TARGET="ghcr.io/${{ github.repository_owner }}/nxdk"
+          else
+            TARGET="ghcr.io/${{ github.repository_owner }}/nxdk-${{ matrix.tag }}"
+          fi
+          docker tag ghcr.io/${{ github.repository }}:${{ matrix.tag }} ${TARGET}:latest
+          docker tag ghcr.io/${{ github.repository }}:${{ matrix.tag }} ${TARGET}:git-${GITHUB_SHA::8}
```

Comment:
Please note using this way will not work as docker require <sub>lower</sub>cases. I had ran into this problem now and before. This change will affect upstream due to XboxDev has couple <sup>upper</sup>cases.

References:
- From GitHub Action
   `Error: buildx failed with: ERROR: invalid tag "ghcr.io/RadWolfie/nxdk-debug:git-62ce1e1a": repository name must be lowercase`
- From docker command directly
   `Error parsing reference: "ghcr.io/RadWolfie/nxdk-debug:latest" is not a valid repository/tag: invalid reference format: repository name must be lowercase`

Reply:
Good catch, will fix this one

Reply:
Fixed

Reply:
Wait, I'm now starting to see the extend of this problem. I did a quick test and this requires more work. Thanks for bringing it up.

Reply:
Got it